### PR TITLE
test: Fix timing edge case in metadata dump test

### PR DESCRIPTION
**Pull request by ralcolea**
PR #412 missed handling an edge case where the metadata dump could begin precisely as the test's validation phase starts. In such cases, the expected log entries may not be available in time, leading to random test failures.

This commit introduces a synchronization step to ensure the test always begins in a consistent state, even when the dump is triggered at the boundary condition described.

### CODE REVIEW

File: tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh [hunk 1]
```diff
@@ -22,39 +25,62 @@ done
...
-		sleep 1
-	done
+# If metadata dump is already in progress, we add a delay to sync up with the next one
+if is_time_for_metadata_dump; then
+	echo "Metadata dump is already in progress, waiting 2 seconds to sync up with the next one"
+	sleep 2
+fi
+
+# Wait for the next metadata dump execution
+truncate -s0 "${TEMP_DIR}/log"
```
[the remark this answers is not in the document]
Comment (ralcolea) [reply tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh]: I liked your suggestion. I applied it :)

File: tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh [hunk 2]
```diff
@@ -22,39 +25,62 @@ done
...
+
 # Confirm metadata dump was not skipped in the master
-assert_awk_finds_no '/metadata dump was skipped:/' "${TEMP_DIR}/log"
+log=$(cat "${TEMP_DIR}/log")
+truncate -s0 "${TEMP_DIR}/log"
+assert_awk_finds_no '/metadata dump was skipped/' "${log}"
 
 # Confirm metadata dump was successful in the master and shadow
-assert_eventually "grep 'periodic metadata dump: success' ${TEMP_DIR}/log | wc -l" "2"
+assert_awk_finds '/periodic metadata dump:/' "${log}"
```
[the remark this answers is not in the document]
Comment (ralcolea) [reply tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh]: You're right. This was the initial intention of the test.
I added your suggestion, but using `grep` instead of `awk`.

File: tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh [hunk 3]
```diff
@@ -1,9 +1,12 @@
 timeout_set '1 minutes'
 period=10
+wait_period=3
```
[the remark this answers is not in the document]
Comment (ralcolea) [reply tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh:3]: The value was experimentally validated considering the workload and operations performed during the test.

File: tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh [hunk 4]
```diff
@@ -22,39 +25,62 @@ done
...
+# Wait for the next metadata dump execution
+truncate -s0 "${TEMP_DIR}/log"
+while ! is_time_for_metadata_dump; do
+	echo "Waiting for next metadata dump, current seconds: $(date +%S)"
+	sleep 1
 done
 
+# It's time for the metadata dump, wait a bit more for the master to finish it
+echo "Time for metadata dump, waiting ${wait_period} seconds, current seconds: $(date +%S)"
+sleep ${wait_period}
```
[the remark this answers is not in the document]
Comment (ralcolea) [reply tests/test_suites/ShortSystemTests/test_metadata_prevent_dump_and_chunks_registration.sh]: Your suggestion is good. Maybe in the future the test can be updated :)

**Review by lgsilva3087 — APPROVED** [review]
LGTM

**Comment by uristdwarf** [comment]
@ralcolea Maybe it's better to wait until after the first metadata dump should occur before starting master? I can still see some timing issues with this (if you are quite unlucky).

Alternatively, you could use libfaketime to manipulate the time

**Comment by ralcolea** [comment]
> @ralcolea Maybe it's better to wait until after the first metadata dump should occur before starting master? I can still see some timing issues with this (if you are quite unlucky).
> 
> Alternatively, you could use libfaketime to manipulate the time

Thank you for the advice @uristdwarf.
I think the time is not anymore.
The issue was the number of dumps in the last part of the test (after chunks registration finishes).

The test validated that exactly 2 dumps (one for each MDS) occurred before the end of the test.
However, if the test tooks longer to finish, more than 2 dumps can occur, like 4 dumps, for instance.
This was validated in the last commit. Hopefully, all the scenarios are already covered :)

**Review by dmga44 — APPROVED** [review]
